akka-actor/src/main/scala/akka/util/MessageBuffer.scala

```diff
@@ -109,13 +110,49 @@ final class MessageBuffer private (private var _head: MessageBuffer.Node, privat
    * @param f the function to apply to each element
    */
   def forEach(f: Procedure2[Any, ActorRef]): Unit = foreach { case (message, ref) => f(message, ref) }
+
+  /**
+   * INTERNAL API
+   */
+  @InternalApi private[akka] def filterNot(p: (Any, ActorRef) => Boolean): Unit = {
+    // easiest to collect a new list, and then re-link the nodes
+    var result = Vector.empty[Node]
+    var node = _head
+    while (node ne null) {
+      if (!p(node.message, node.ref)) {
+        result :+= node
+      }
+      node = node.next
+    }
+
+    _size = result.size
+    if (_size == 0) {
+      _head = null
+      _tail = null
+    } else if (_size == 1) {
+      _head = result.head
+      _tail = result.head
+      _tail.next = null
+    } else {
+      _head = result.head
+      _tail = result.last
+      _tail.next = null
+      result.zipWithIndex.foreach {
```

Comment:
remove the zipWithIndex with plain while?

Reply:
changed to while